### ISSUE TITLE
Layouts with Same Field Type Duplicates Content 

### ISSUE THREAD

**Issue by jeffolmstead** [OrchardCMS/Orchard#5974]
I have created a movie content type which has three taxonomy fields (e.g. Genre, Rating, Program) and two media library picker fields (e.g. Image, Trailer) and placed all of these fields into a Layout. I then hid all of their displays (except for under the "Layout" Display Type). 

I then created a new content item having values for all three taxonomy fields and for both media library picker fields. What I ended up with was 3 sets of all three taxonomy fields and 2 sets of all two media library picker fields. 

In short, the Layout module appears to not differentiate between field names, it processes all Media Library Pickers in one batch (2 times since I had two fields) and all Taxonomy Fields in one batch (3 times since I had three fields). I believe the issue lies in Orchard.Layouts.Services.ContentFieldDisplay ==> BuildDisplay:

```
public dynamic BuildDisplay(IContent content, ContentField field, string displayType, string groupId) {
            var context = BuildDisplayContext(content, displayType, groupId);
            var drivers = GetFieldDrivers(field.FieldDefinition.Name);
```
 The provided context has no bearing on the particular field and the driver is only based on the field definition name. This means all fields of the same type are processed. Aside from the performance issues, I also I see no place where the particular result for only the desired field is separated back out. 

It almost seems that processing in Orchard.Layouts.Providers.ContentFieldElementHarvester ==> Displaying should be grouped by field definition name so the driver is only called once then the results disseminated (likely a lot harder than seems). 

Look forward to other thoughts on how to resolve this.

**Comment by jeffolmstead** [comment]
I have made the following changes in Orchard.Layouts.Services.ContentFieldDisplay and it does seem to fix the problem. Here is the original:

```
        public dynamic BuildDisplay(IContent content, ContentField field, string displayType, string groupId) {
            var context = BuildDisplayContext(content, displayType, groupId);
            var drivers = GetFieldDrivers(field.FieldDefinition.Name);

            drivers.Invoke(driver => {
                var result = driver.BuildDisplayShape(context);
                if (result != null) {
                    result.Apply(context);
                }
            }, Logger);

            return context.Shape;
        }
```
And here it is with the changes implemented
```
        public dynamic BuildDisplay(IContent content, ContentField field, string displayType, string groupId) {
            var context = BuildDisplayContext(content, displayType, groupId);
            var drivers = GetFieldDrivers(field.FieldDefinition.Name);

            drivers.Invoke(driver => {
                var result = driver.BuildDisplayShape(context);
                if (result != null) {
                    if (result is CombinedResult) {
                        var cResults = ((CombinedResult)result).GetResults();
                        foreach (var cResult in cResults) {
                            if (cResult.ContentField.Name == field.Name) {
                                cResult.Apply(context);
                            }
                        }
                    }
                    else {
                        if (result.ContentField.Name == field.Name) {
                            result.Apply(context);
                        }
                    }
                }
            }, Logger);

            return context.Shape;
        }
```
In short, we are only applying the active field we are interested in to the context, not every field of the same type. See any trickle down issues?

**Comment by sfmskywalker** [comment]
Fixed in 3a7091467b19e468fe59346c1dcfd725e3e3888e.
